ang/crmMailing/services.js

```diff
@@ -60,17 +60,23 @@
     window.tpls = tpls;
     var lastModifiedTpl = null;
     return {
+      // Get a template
+      // @param id MessageTemplate id  (per APIv3)
       // @return Promise MessageTemplate (per APIv3)
       get: function get(id) {
-        id = '' + id; // parseInt(id);
         var dfr = $q.defer();
-        var tpl = _.where(tpls, {id: id});
-        if (id && tpl && tpl[0]) {
-          dfr.resolve(tpl[0]);
-        }
-        else {
-          dfr.reject(id);
-        }
+        var tpl = CRM.api3('MessageTemplate', 'get', {
```

Comment:
From IRC discussion - this section could probably be simplified. This code is basically converting the jQuery-style promise (returned by CRM.api3) into an angular-style promise. The [`crmApi()`](https://github.com/civicrm/civicrm-core/blob/4.6/ang/crmUtil.js#L14) function will do that. If you call `crmApi('MessageTemplate','getsingle',...)`, then you can probably delete most of the `dfr` stuff.